.github/workflows/makefile.yaml

```diff
@@ -0,0 +1,63 @@
+name: Makefile
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: docker://ghcr.io/pulumi/devcontainer:latest
+      options: --user root
+    permissions:
+      contents: read
+      packages: read
+      actions: read
+    steps:
+    -
+      name: Checkout repository
+      uses: actions/checkout@v4
+      id: git
+      with:
+        submodules: 'recursive'
+    -
+      name: Unshallow clone for tags
+      id: tags
+      run: |
+        set -ex
+        sudo chown -R $(whoami) /__w/pulumi-provider-boilerplate/pulumi-provider-boilerplate
+        git config --global --add safe.directory /__w/pulumi-provider-boilerplate/pulumi-provider-boilerplate
```

Comment:
Why do we need to set this on each run? We don't do this in other providers.

Reply:
I found this command `git config --global --add safe.directory` to be necessary once running the workflow inside a non-root container.

Reply:
An alternate resolution would be to run the container as root inside the container itself.